corehq/apps/reports/util.py

```diff
@@ -531,3 +535,60 @@ def update_tableau_user(domain, username, role=None, groups=[]):
     user.save()
     for group in groups:
         session.add_user_to_group(user.tableau_user_id, group.id)
+
+
+@periodic_task(run_every=crontab(minute=0, hour='*/1'), queue='background_queue')
+def sync_all_tableau_users():
+    def _sync_tableau_users_with_hq(domain):
+        tableau_user_names = [tableau_user.username for tableau_user in TableauUser.objects.filter(
+            server=TableauServer.objects.get(domain=domain)
+        )]
+        web_users_names = [web_user.username for web_user in WebUser.by_domain(domain)]
+        # If there's a web user that isn't in the TableauUser model, create a new Tableau user
+        for web_user_name in web_users_names:
+            if web_user_name not in tableau_user_names:
+                _add_tableau_user_local(domain, web_user_name)
+        # If there's a TableauUser with no corresponding WebUser, delete the Tableau user
+        for tableau_user_name in tableau_user_names:
+            if tableau_user_name not in web_users_names:
+                _delete_user_local(domain, tableau_user_name)
+
+    def _sync_tableau_users_with_remote(domain):
+        remote_groups = get_all_tableau_groups(domain)
+        session = TableauAPISession.create_session_for_domain(domain)
+
+        remote_HQ_group = None
+        for group in remote_groups:
+            if group.name == 'HQ':
+                remote_HQ_group = group
+                remote_groups.remove(group)
```

Comment:
What's the purpose of this line. Also this will raise an exception since you are modifying the list while iterating over it